hub/util/transform.py

```diff
@@ -172,14 +179,18 @@ def _transform_sample_and_update_chunk_engines(
     result = result_resolved  # type: ignore
     result_keys = set(result.keys())
 
+    # replace temporary tensors with actual ones for comparing
+    tensors_ = (set(tensors) - set(label_temp_tensors.values())).union(
```

Comment:
we should probably do this once outside of here and reuse tensors_ instead of doing for every sample